src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state }; // Create a copy of the initial state
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      currentState = {
+        ...currentState, ...action.extraData,
+      };
+    } else if (action.type === 'removeProperties') {
+      // Create a new state object with specified properties removed
+      const { keysToRemove } = action;
+
+      for (const key of keysToRemove) {
+        if (currentState.hasOwnProperty(key)) {
+          delete currentState[key];
+        }
+      }
+    } else if (action.type === 'clear') {
+      // Create an empty state object
+      currentState = {};
+    }
+
+    // Create a clone of the current state and add it to the result array
+    result.push({ ...currentState });
```

Comment:
Good work but in this case i'd recommend to use switch/case statement as we rely on single action.type value

Reply:
Better now?